fl_training.py

```diff
@@ -21,52 +21,44 @@
 
 #%% Pre-process data for ML training
 
-
-def preprocess_node_list(node_list):
-    """Return node_list preprocessed for keras CNN"""
-
-    print("\n## Pre-processing train data of each node for keras CNN:")
-    for node_index, node in enumerate(node_list):
-
-        # Preprocess input (x) data
-        node.preprocess_data()
-
+def preprocess_scenarios_data(scenario):
+    """Return scenario with central datasets (valearlystop, test) and distributed datasets (nodes) pre-processed"""
+
+    print("\n## Pre-processing datasets of the scenario for keras CNN:")
+
+    # First, datasets of each node
+    for node_index, node in enumerate(scenario.node_list):
+        # Preprocess inputs (x) data
+        node.x_train = utils.preprocess_input(node.x_train)
+        node.x_test = utils.preprocess_input(node.x_test)
+        # Preprocess labels (y) data
+        node.y_train = keras.utils.to_categorical(node.y_train, constants.NUM_CLASSES)
+        node.y_test = keras.utils.to_categorical(node.y_test, constants.NUM_CLASSES)
         # Crete validation dataset
-        x_node_train, x_node_val, y_node_train, y_node_val = train_test_split(
+        node.x_train, node.x_val, node.y_train, node.y_val = train_test_split(
```

Comment:
Note: for now we only use `node.x_val` for plotting figures
However I think we should keep it that way for now and it might be useful when we make our federated learning approcach a bit smarter.

Reply:
OK